### PR TITLE
Docker for Redis Stack

### PR DESCRIPTION
**Pull request by aosan**
### Background

The aim of this PR is to make the use of Redis Stack beginner friendly and help developers get consistent results.
Developers and curious people can access RedisInsight at  http://localhost:8001/ to see the interaction between Auto-GPT and Redis caching engine.

### Changes

Made it easier to use Redis by providing the steps in the updated documentation.
Consistent build, configuration and testing using a docker-compose.yml type of configuration.
Easy start of the Docker container by using a shell script.
Easier debugging and tracing by using Redis Stack with built-in RedisInsight for development cycles.
Better security by enabling an access password for "Production Ready".
 
- Updated README.md with a Docker for Linux.
- Added the Docker Compose file as `docker_redis.yml`
- Added a shell script to make it easy to start the container with `start_redis_docker.sh`

### Test Plan

Configuration validation with `docker-compose -f docker_redis.yml config` on:
- Fedora Linux 37 with Docker version 23.0.3, build 3e7cbfd and Docker Compose version v2.17.2.
- Third party confirmation of correct build on Windows 11 with Docker Desktop 4.18.0 and Docker Compose version v2.17.2.
Multiple build cycles with `docker system prune -a` to eliminate any image conflicts potential

Successful configuration and built of a Docker Stack image and confirmed:
- .env has the correct Redis configuration
- Running Auto-GPT reports “Using memory of type: RedisMemory”
- Checking http://localhost:8001/redis-stack/browser and confirming the CPU, Total Memory, Total Keys and Connected Clients all report positive non-zero values.

### Change Safety

- [x] I have added tests to cover my changes
- [x] I have considered potential risks and mitigations for my changes

### CODE REVIEW

**Review by nponeccop — CHANGES_REQUESTED** [review]
Split it into 3 atomic PRs:

- Docker for Redis (and corresponding README changes)
- Unrelated README changes
- CONTRIBUTING changes

File: README.md [hunk 1]
```diff
@@ -143,7 +147,7 @@ export CUSTOM_SEARCH_ENGINE_ID="YOUR_CUSTOM_SEARCH_ENGINE_ID"
 
 ## Redis Setup
 
-Install docker desktop.
+Install Docker Desktop.
```
Comment (ksylvan) [inline README.md:150]: Many people aren't using Docker Desktop for OCI compliant containers these days.

I've tested this with Rancher Desktop and it works with the "containerd" engine and "nerdctl" as well. You simply have to change the docker-compose line to "nerdctl compose"

**Comment by nponeccop** [comment]
As for Docker Desktop, does Rancher Desktop work on Windows? We mostly care about Windows folks, the Linux folks usually know what they are doing and can choose their container engine at will.

**Comment by ksylvan** [comment]
> As for Docker Desktop, does Rancher Desktop work on Windows?

Yes. I use it daily on both Windows 10 and on Mac OS.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Boostrix** [comment]
this really looks like it could be very useful once it can be reviewed/integrated - Docker/redis really need some TLC :-)

**Comment by Pwuts** [comment]
This PR is very much out of date, and Redis has been removed from the default docker config

**Pull request closed without merging**